bin/asl-say

```diff
@@ -145,6 +164,29 @@ if [ -z "${NODE}" ] || [ ${NODE} -lt 1 ]; then
 	exit
 fi
 
+if [ -n "${ALT_DATE}${ALT_TIME}" ]; then
+	if [ -n "${ALT_DATE}" ]; then
+		re=^[0-9]{8}$
+		if ! [[ ${ALT_DATE} =~ $re ]]; then
+			echo "Alternate date must be YYYYMMDD"
+			exit 1
+		fi
+	else
+		ALT_DATE=$(date +%Y%m%d)
+	fi
+	if [ -n "${ALT_TIME}" ]; then
+		re=^[0-9]{4}$
+		if ! [[ ${ALT_TIME} =~ $re ]]; then
+			echo "Alternate time must be HHMM"
+			exit 1
+		fi
+	else
+		ALT_TIME=$(date +%H%M)
+	fi
```

Comment:
By forcing these regexes, we lose a lot of functionality of the `date` command. For instance, I can put `date -d "1 week"` to print a datetime 7 days from now. Or `date -d "-15 minutes"` to print the time it was 15 minutes ago.
Not sure how many people would actually use this, but it seems easier to rely on the existing `date` functionality that users might be familiar with, and it doesn't limit the advanced users from doing more powerful things with the command. For instance you could just check the return value of `date -d ${ALT_DATE_TIME}` and if it fails stop the script there. (See comment above about combining date and time into one input argument).

Reply:
Again, my macOS system does not support `date -d`.   I shall be removing most of my alternate date/time hacks.